src/compiler/checker.ts

```diff
@@ -6699,7 +6699,14 @@ namespace ts {
                 let hasThisParameter: boolean;
                 const iife = getImmediatelyInvokedFunctionExpression(declaration);
                 const isJSConstructSignature = isJSDocConstructSignature(declaration);
-                const isUntypedSignatureInJSFile = !iife && !isJSConstructSignature && isInJavaScriptFile(declaration) && !hasJSDocParameterTags(declaration);
+                const isUntypedSignatureInJSFile = !iife &&

```

Comment:
What case is handling when the declaration is an object literal with construct signatures, (ie, `{(): void}`)?

Reply:
`!getJSDocType(declaration)`. I'll add a test case. Note that the new code in `getJSDocType` doesn't look deep enough to get type of the parameter out of the method declaration inside the object literal type.